app/assets/javascripts/admin/addon/controllers/admin-user-index.js

```diff
@@ -626,5 +559,57 @@ export default Controller.extend(CanCheckEmails, {
         }
       });
     },
+
+    showDeletePostsConfirmation() {
+      showModal("admin-delete-posts-confirmation", {
+        admin: true,
+        model: this.model,
+      });
+    },
+
+    deleteAllPosts() {
+      let deletedPosts = 0;
+      let deletedPercentage = 0;
+      const user = this.model;
+
+      const performDelete = (progressModal) => {
+        this.model
+          .deleteAllPosts()
+          .then(({ posts_deleted }) => {
+            if (posts_deleted === 0) {
+              user.set("post_count", 0);
+              progressModal.send("closeModal");
+            } else {
+              deletedPosts += posts_deleted;
+              deletedPercentage = Math.floor(
+                (deletedPosts * 100) / user.get("post_count")
+              );
+              progressModal.setProperties({
+                deletedPercentage: deletedPercentage,
+              });
+              performDelete(progressModal);
+            }
+          })
+          .catch((e) => {
+            progressModal.send("closeModal");
+            let error;
+            AdminUser.find(user.get("id")).then((u) => user.setProperties(u));
+            if (e.jqXHR.responseJSON && e.jqXHR.responseJSON.errors) {
+              error = e.jqXHR.responseJSON.errors[0];
+            }
+            error = error || I18n.t("admin.user.delete_posts_failed");
+            bootbox.alert(error);
+          });
+      };
+
+      const openProgressModal = () => {
```

Comment:
I don't understand why you are making a closure here and then calling it immediately afterwards. Could you not set `const progressModal = showModal("admin-delete-user-posts-progress", { admin: true })` as one line?

Reply:
I am not sure why I did that, I was trying to reuse the original code but forgot to optimize it. I've now made the change you suggested.